sonarts-sq-plugin/its/plugin/src/test/java/org/sonar/typescript/its/RuleConfigurationTest.java

```diff
@@ -0,0 +1,188 @@
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.zip.ZipEntry;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.rule.ActiveRules;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.rule.Checks;
+import org.sonar.api.batch.rule.internal.ActiveRulesBuilder;
+import org.sonar.api.batch.rule.internal.NewActiveRule;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.FileLinesContext;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.plugin.typescript.ExternalProcessStreamConsumer;
+import org.sonar.plugin.typescript.ExternalTypescriptSensor;
+import org.sonar.plugin.typescript.TypeScriptLanguage;
+import org.sonar.plugin.typescript.TypeScriptRules;
+import org.sonar.plugin.typescript.TypeScriptRulesDefinition;
+import org.sonar.plugin.typescript.executable.ExecutableBundle;
+import org.sonar.plugin.typescript.executable.ExecutableBundleFactory;
+import org.sonar.plugin.typescript.executable.SonarTSCoreBundle;
+import org.sonar.plugin.typescript.executable.Zip;
+import org.sonar.plugin.typescript.rules.TypeScriptRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+@RunWith(Parameterized.class)
+public class RuleConfigurationTest {
+
+  private static final Path RULE_LINT_FILES = Paths.get("../../../sonarts-core/tests/rules");
+  private static final Pattern ERROR_COMMENT = Pattern.compile("//.*\\^+[^<>]*");
+
+  @ClassRule
+  public static TemporaryFolder temp = new TemporaryFolder();
+
+  private static File baseDir;
+  private SensorContextTester testContext;
+  private static ExternalProcessStreamConsumer errorConsumer;
+  private static ExecutableBundleFactory executableBundleFactory;
+  private static FileLinesContextFactory fileLinesContextFactory;
+  private static NoSonarFilter noSonarFilter;
+
+  @Parameter
+  public String ruleKey;
+
+  @Parameter(1)
+  public String lintFile;
+
+  @Parameters(name = "{0}:{1}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+      {"S3776", "cognitiveComplexityRule"},
```

Comment:
@saberduck in future this hardcoded list might be requested from TypeScriptRules, right?